utils/earth_utils.py

```diff
@@ -203,3 +203,32 @@ def calculate_mgrs_zones(latitudes: np.ndarray, longitudes: np.ndarray) -> np.nd
     mgrs_regions[valid_indices] = mgrs_regions_flat
 
     return mgrs_regions
+
+def noisy_bearing_measurement(vec: np.ndarray, sigma=np.sqrt(0.001)):
+    """
+    Add Gaussian noise to a bearing measurement.
+    Parameters:
+        vec (np.ndarray): The original bearing vector.
+        sigma (float): The standard deviation of the noise.
+
+    Returns:
+        np.ndarray: The noisy bearing vector.
+    """
+    # Choose an arbitrary vector that isn't parallel to vec
+    if np.allclose(vec, [0, 0, 1]):
```

Comment:
Instead of using `np.allclose` which has a tight tolerance by default, we can just the basis vector with a smaller dot product with `vec`.
```suggestion
    if abs(vec[0]) <= abs(vec[2]):
```